postgres.tf

```diff
@@ -134,7 +140,7 @@ resource "azurerm_virtual_network_peering" "dg_postgres" {
 # connect data gateway vnet to private dns zone (this will contain the A name for postgres)
 resource "azurerm_private_dns_zone_virtual_network_link" "postgres_dg" {
   name                  = format("%s-%s-virtual-network-link", var.product, var.env)
-  resource_group_name   = var.DNSResGroup
+  resource_group_name   = data.azurerm_resource_group.privatelink_resource_group.name
```

Comment:
whats current issue? this is just swapping string to data return rather than static variable

Reply:
yes, I was trying to fix the issue where 2 resource groups have the same name but couldn't fix